dsa-23au/python-dsa/wshine-mst/tests/test_mst.py

```diff
@@ -0,0 +1,83 @@
+from mst import prims_alg, Edge, Graph
+import os
+edges = [
+    ("RN", "KB", 24),
+    ("RN", "CP", 22),
+    ("KB", "CP", 4),
+    ("SB", "KB", 8),
+    ("RG", "KB", 7),
+    ("SB", "RG", 10),
+    ("RG", "IGA", 14),
+    ("RG", "CK", 12),
+    ("CK", "KB", 10),
+    ("CK", "CP", 6),
+    ("IGA", "KB", 19),
+    ("IGA", "CP", 20),
+    ("IGA", "CK", 16),
+    ("IGA", "M", 14),
+    ("M", "GK", 12),
+    ("M", "CK", 15),
+    ("GK", "CK", 14),
+    ("GK", "CP", 17)
+]
+
+
+def test_acyclic():
```

Comment:
nice work Winston, it is indeed non-obvious that an MST does not contain any cycles. It's always good to instrument your code with additional checks like this, and make them easy to test, to convince yourself you understand what's going on.